### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by Gaurav0**
This should be considered a breaking change.

### CODE REVIEW

**Comment by Turbo87** [comment]
what caused these test failures?

**Comment by Gaurav0** [comment]
I'm not sure why, but .eslintrc.js is no longer being linted after the broccoli updates.

**Comment by Gaurav0** [comment]
In any case, https://github.com/eslint/eslint/issues/8429

**Comment by rwjblue** [comment]
FWIW, `.eslintrc.js` is a default ignored file AFAICT...

File: test/eslint-test.js [hunk 1]
```diff
@@ -161,7 +161,7 @@ describe('broccoli-lint-eslint', function() {
       yield output.build();
 
       let result = output.read();
-      expect(Object.keys(result)).to.deep.equal(['.eslintrc.lint-test.js', 'a.lint-test.js', 'b.lint-test.js']);
+      expect(Object.keys(result)).to.deep.equal(['.eslintrc.js', 'a.lint-test.js', 'b.lint-test.js']);
```
Comment (rwjblue) [inline test/eslint-test.js]: I do find this curious... Why would these dep updates have broken this? 🤔

Comment (Gaurav0) [reply test/eslint-test.js]: I thought it was because .eslintrc.js wasn't being linted any more.

File: yarn.lock [hunk 2]
```diff
@@ -669,173 +598,71 @@ detect-file@^0.1.0:
...
-    text-table "~0.2.0"
-    user-home "^2.0.0"
+eslint-visitor-keys@^1.0.0:
+  version "1.0.0"
+  resolved "https://registry.yarnpkg.com/eslint-visitor-keys/-/eslint-visitor-keys-1.0.0.tgz#3f3180fb2e291017716acb4c9d6d5b5c34a6a81d"
 
 eslint@^4.11.0:
-  version "4.11.0"
-  resolved "https://registry.yarnpkg.com/eslint/-/eslint-4.11.0.tgz#39a8c82bc0a3783adf5a39fa27fdd9d36fac9a34"
+  version "4.19.1"
```
Comment (rwjblue) [inline yarn.lock]: Why is this bumped? I suspect this is why the `.eslintrc.js` file is now not processed....

https://github.com/ember-cli/broccoli-lint-eslint/blob/75aec8f27a817b39e3ab707430b24766e78d2294/lib/index.js#L137

Most likely between eslint@4.11.0 and 4.19.1 `isPathIgnored` started returning `true` for `.eslintrc.js`...

Comment (Gaurav0) [reply yarn.lock]: I did run yarn...

File: test/eslint-test.js (hunk 1 above)
Comment (Turbo87) [reply test/eslint-test.js]: but you're updating broccoli dependencies. those updates shouldn't influence what is being linted or not 🤔

File: yarn.lock (hunk 2 above)
Comment (Turbo87) [reply yarn.lock]: ah, that might indeed be the issue. I forgot to look at the lockfile 🤦‍♂️

Comment (Gaurav0) [reply yarn.lock]: We are about to bump it anyway in the next PR.

Comment (rwjblue) [reply yarn.lock]: I don't see an obvious culprit in the release notes of those versions, but https://github.com/eslint/eslint/issues/10341 does seem related.

Comment (Turbo87) [reply yarn.lock]: it's still a change that's quite unrelated to bumping broccoli dependencies, which this PR is apparently about

Comment (rwjblue) [reply yarn.lock]: I also created https://github.com/ember-cli/ember-cli/issues/8070 to fix the new project blueprint.

Comment (Gaurav0) [reply yarn.lock]: So I can change the name of the PR to bumping dependencies, lol.

File: test/eslint-test.js (hunk 1 above)
Comment (Gaurav0) [reply test/eslint-test.js]: So, I think @rwjblue is right, it's due to the change in ESLint version. The new version is in the semver range, so existing users are getting it when they install the add-on already.

Comment (rwjblue) [reply test/eslint-test.js]: right, seems fine, but there is (AFAICT) no reason the lock file should have updated eslint versions here

I'd suggest running `git checkout master yarn.lock && yarn`, which should reset the `yarn.lock` to whatever master has and _only_ update the deps that you explicitly changed in the package.json.

In addition, we should double check to confirm that the `.travis.yml` is using the lockfile (vs using `--no-lockfile`).

Comment (Gaurav0) [reply test/eslint-test.js]: thanks!

Comment (Gaurav0) [reply test/eslint-test.js]: done!

**Comment by Gaurav0** [comment]
@rwjblue @Turbo87 Can we merge this now?

**Comment by Turbo87** [comment]
yeah, was just waiting for CI to finish

**Pull request merged**